### DISCUSSION TITLE
Sorting data based on one-to-many relationship

### DISCUSSION THREAD

**Discussion by zeval** [prisma/prisma#13045] [Q&A] [answered]
I'm using Prisma with a MySQL database for a project, an online shop, where the same product can be sold by multiple suppliers. 

At a certain part of my database I have a table structure such as this one:
* Product (id, name, ...)
* Supply (productId, supplier, price, stock, ...)

The `Supply` table represents the fact that a supplier sells a certain product. Obviously, a `Product` can have multiple `Supply` entries, as many as the number of suppliers that sell said product. 

I'm looking to use Prisma's `findMany` operation to sort products by the lowest price out of all `Supplies` a product has. I can easily calculate this value after obtaining all products, and sort in that way, but that wouldn't be scalable, and I'm also using pagination with `take` and `skip` for efficiency purposes, rendering that alternative pointless. 

As an example, say two `Product` products have two `Supplies` each:
* Product X, sold by supplier A at 63€ and by supplier B at 75€
* Product Y, sold by supplier B at 54€ and by supplier A at 60€

The natural result I'm looking for (when sorting in ascending order) would be:
1. Product Y
2. Product X

As product X's minimum `Supply` price is lower than product Y's minimum `Supply` value.   

I've thoroughly researched Prisma's documentation regarding this situation and cannot find a solution. I would very much like to implement this using the `orderBy` property for the `findMany` operation instead of using raw SQL queries, as I have implemented many other types of sorting the Prisma way.

Any help regarding this issue is appreciated, thanks in advance.

**Comment by austincrim** [comment, marked as answer]
Hey @zev4l 👋 ,

I don't believe what you are wanting is possible yet without the usage of `$queryRaw`. If you could submit a detailed [feature request](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=), our engineers will take it from there!

One idea I had that may or may not be useful:
```ts
let result = await prisma.supply.groupBy({
  by: ['productId'],
  orderBy: {
    _min: {
      price: 'asc'
    }
  }
})
```
This query won't return the full `Product`, but it should get you a list of `productIds` sorted from least to most expensive.

**Reply by zeval** [reply]
Hey @austincrim!

Thanks a lot! That's looking at the problem from a completely different perspective! I appreciate it a lot, it's already a step forward. I have big function which dynamically specifies filters, other sorting arguments, offset pagination, full-text search, etc. Is there any way I can have findMany follow this list of IDs?

Regarding the feature request, I'll definitely be writing one, though I have yet to detail exactly what this situation is. Thanks again!

**Reply by austincrim** [reply]
You will have to loop over the ID's and create a query for each of them. 
```ts
let products = await Promise.all(
  productIds.map((id) => prisma.product.findUnique({ where: { id } }))
)
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
